src/V3SchedReplicate.cpp

```diff
@@ -0,0 +1,256 @@
+
+        if (senTreep->hasClocked()) {
+            // Clocked logic is never triggered by reads
+            readTriggersThisLogic = [](AstVarScope*) { return false; };
+        } else if (senTreep->hasCombo()) {
+            // Combinational logic is always triggered by reads
+            readTriggersThisLogic = [](AstVarScope*) { return true; };
+        } else {
+            UASSERT_OBJ(senTreep->hasHybrid(), activep, "unexpected");
+            // Hybrid logic is triggered by all reads, except for reads of the explicit
+            // sensitivities
+            readTriggersThisLogic = [](AstVarScope* vscp) { return !vscp->user4(); };
+            senTreep->foreach<AstVarRef>([](const AstVarRef* refp) {  //
+                refp->varScopep()->user4(true);
+            });
+        }
+
+        for (AstNode* nodep = activep->stmtsp(); nodep; nodep = nodep->nextp()) {
+            LogicVertex* const lvtxp = new LogicVertex{graphp, scopep, senTreep, nodep, region};
+            const VNUser2InUse user2InUse;
+            const VNUser3InUse user3InUse;
+
+            nodep->foreach<AstVarRef>([&](AstVarRef* refp) {
+                AstVarScope* const vscp = refp->varScopep();
+                VarVertex* const vvtxp = getVarVertex(vscp);
+
+                // If read, add var -> logic edge FIXMEV5: explain the third term in the condition
+                if (refp->access().isReadOrRW() && !vscp->user3SetOnce()
+                    && readTriggersThisLogic(vscp) && !vscp->user2()) {  //
+                    addEdge(vvtxp, lvtxp);
+                }
+                // If written, add logic -> var edge
+                if (refp->access().isWriteOrRW() && !vscp->user2SetOnce()) {  //
+                    addEdge(lvtxp, vvtxp);
+                }
+            });
+        }
+    };
+
+    logicRegions.m_pre.foreach ([&](AstScope* scopep, AstActive* activep) {  //
+        addLogic(ACTIVE, scopep, activep);
+    });
+    logicRegions.m_active.foreach ([&](AstScope* scopep, AstActive* activep) {  //
+        addLogic(ACTIVE, scopep, activep);
+    });
+    logicRegions.m_nba.foreach ([&](AstScope* scopep, AstActive* activep) {  //
+        addLogic(NBA, scopep, activep);
+    });
+
+    return std::unique_ptr<Graph>{graphp};
+}
+
+void propagateDrivingRegions(Graph* graphp) {
+    // FIXMEV5: better algorithm (the graph is guaranteed to be acyclic at this point,
+    // so iterate in breadth first (rank) order
+
+    bool changed;
+    do {
+        changed = false;
+        // For each vertex
+        for (V3GraphVertex* gvtxp = graphp->verticesBeginp(); gvtxp;
+             gvtxp = gvtxp->verticesNextp()) {
+            Vertex* const vtxp = static_cast<Vertex*>(gvtxp);
+            // Based on inputs, compute union of all driving regions
+            uint8_t drivingRegions = 0;
+            for (V3GraphEdge* edgep = vtxp->inBeginp(); edgep; edgep = edgep->inNextp()) {
+                Vertex* const srcp = static_cast<Vertex*>(edgep->fromp());
```

Comment:
```suggestion
            const Vertex* const vtxp = static_cast<Vertex*>(gvtxp);
            // Based on inputs, compute union of all driving regions
            uint8_t drivingRegions = 0;
            for (V3GraphEdge* edgep = vtxp->inBeginp(); edgep; edgep = edgep->inNextp()) {
                const Vertex* const srcp = static_cast<Vertex*>(edgep->fromp());
```

Reply:
Only the second one is correct, but done.